web3/providers/ipc.py

```diff
@@ -42,22 +42,65 @@ def get_default_ipc_path(testnet=False):
         testnet = ""
 
     if sys.platform == 'darwin':
-        return os.path.expanduser(os.path.join(
+        ipc_path = os.path.expanduser(os.path.join(
             "~",
             "Library",
             "Ethereum",
             testnet,
-            "geth.ipc",
+            "geth.ipc"
         ))
+        if os.path.exists(ipc_path):
+            return ipc_path
+
+        ipc_path = os.path.expanduser(os.path.join(
+            "~",
+            "Library",
+            "Application Support",
+            "io.parity.ethereum",
+            "jsonrpc.ipc"
+        ))
+        if os.path.exists(ipc_path):
+            return ipc_path
+
     elif sys.platform.startswith('linux'):
-        return os.path.expanduser(os.path.join(
+        ipc_path = os.path.expanduser(os.path.join(
             "~",
             ".ethereum",
             testnet,
-            "geth.ipc",
+            "geth.ipc"
+        ))
+        if os.path.exists(ipc_path):
+            return ipc_path
+
+        ipc_path = os.path.expanduser(os.path.join(
+            "~",
+            ".local",
+            "share",
+            "io.parity.ethereum",
+            "jsonrpc.ipc"
         ))
+        if os.path.exists(ipc_path):
+            return ipc_path
+
     elif sys.platform == 'win32':
-        return "\\\\.\\pipe\\geth.ipc"
+        ipc_path = os.path.join(
+            "\\\\",
```

Comment:
I so wish we had windows CI to know that this works.... 😢

Reply:
Tagging #166 for motivation.

Reply:
I tested it on my windows machine if it is some consolation to you. I couldn' get seperated `"\\", "\\"` to work. It would merge them to one `\`. That is why I used 4 backslashes.

Reply:
I also had geth and parity running while doing my in house testing, but I agree a more thourough test would be comforting.